code_examples/sdk_examples/src/core_features/messaging/02_encrypt_message.ts

```diff
@@ -0,0 +1,28 @@
+import * as Kilt from '@kiltprotocol/sdk-js'
+import { useEncryptionSignCallback } from '../signCallback/useEncryptionSignCallback'
+
+export async function encryptMessage(
+  message: Kilt.IMessage,
+  senderUri: Kilt.DidUri,
+  receiverUri: Kilt.DidUri,
+  keyAgreement: Kilt.KiltEncryptionKeypair
+): Promise<Kilt.IEncryptedMessage> {
+  const senderDidDocument = await Kilt.Did.resolve(senderUri)
```

Comment:
I am not sure we need to resolve the DID document of the sender. All we need from them is the private encryption key, nothing else. This does not sound like the right approach, since DID resolution only concerns with public information.

Reply:
Is there a way to take the private key and fetch the key uri of the key agreement key? Without fetching the document there doesn't seem to be a nice way of doing this?

Reply:
We have a [`computeKeyId`](computeKeyId) function for that.

Reply:
So you would call that on the public encryption key, and then append it to the DID URI.

Reply:
For me, we should stick to one way of doing it. All our stuff at the moment is complicated enough without having different styles throw into the mix.

Reply:
I sent that in the wrong message. That function is only a `[testUtils](https://github.com/KILTprotocol/sdk-js/blob/c6f38539171f79d5c2ab9e8284fc3d890adf89f1/packages/testing/src/TestUtils.ts#L206)`

Reply:
Yes, we would need to open another ticket to actually have a function the computes the key id.

Reply:
Based on the internal discussion, there does not seem to be a better way than resolving the document right now. Please make a ticket to add the new function to the SDK. I'll mark this comment as resolved.